instrumentation/github.com/aws/aws-sdk-go/service/s3/otels3/s3.go

```diff
@@ -0,0 +1,216 @@
+	}
+
+	s.recorders.operationDuration.Record(
+		spanCtx,
+		float64(time.Since(startTime).Microseconds()),
+		attrs...,
+	)
+	//append spand if and trace id to metrics on latency on gauge
+	s.counters.operation.Add(ctx, 1, attrs...)
+
+	return output, err
+}
+
+// GetObjectWithContext invokes the GetObjectWithContext function with tracing instrumented
+func (s *instrumentedS3) GetObjectWithContext(ctx aws.Context, input *s3.GetObjectInput, opts ...request.Option) (*s3.GetObjectOutput, error) {
+	startTime := time.Now()
+	destination := aws.StringValue(input.Bucket)
+	attrs := createAttributes(destination, operationGetObject)
+
+	spanCtx, span := s.tracer.Start(
+		ctx,
+		fmt.Sprintf("%s.%s", destination, operationGetObject),
+		trace.WithSpanKind(trace.SpanKindClient),
+		trace.WithAttributes(attrs...),
+	)
+
+	output, err := s.S3API.GetObjectWithContext(ctx, input, opts...)
+	callReturnTime := trace.WithTimestamp(time.Now())
+	defer span.End(callReturnTime)
+
+	if err != nil {
+		attrs = append(attrs, labelStatusFailure)
+		span.SetAttributes(labelStatusFailure)
+		span.SetStatus(codes.Error, err.Error())
+	} else {
+		attrs = append(attrs, labelStatusSuccess)
+		span.SetAttributes(labelStatusSuccess)
+		span.SetStatus(codes.Ok, "")
+	}
+
+	if s.spanCorrelation {
+		attrs = appendSpanAndTraceIDFromSpan(attrs, span)
+	}
+
+	s.recorders.operationDuration.Record(
+		spanCtx,
+		float64(time.Since(startTime).Microseconds()),
+		attrs...,
+	)
+	s.counters.operation.Add(ctx, 1, attrs...)
+
+	return output, err
+}
+
+// DeleteObjectWithContext invokes the DeleteObjectWithContext function with tracing instrumented
+func (s *instrumentedS3) DeleteObjectWithContext(ctx aws.Context, input *s3.DeleteObjectInput, opts ...request.Option) (*s3.DeleteObjectOutput, error) {
+	startTime := time.Now()
+	destination := aws.StringValue(input.Bucket)
+	attrs := createAttributes(destination, operationDeleteObject)
+
+	spanCtx, span := s.tracer.Start(
+		ctx,
+		fmt.Sprintf("%s.%s", destination, operationDeleteObject),
+		trace.WithSpanKind(trace.SpanKindClient),
+		trace.WithAttributes(attrs...),
+	)
+
+	output, err := s.S3API.DeleteObjectWithContext(ctx, input, opts...)
+	callReturnTime := trace.WithTimestamp(time.Now())
+	defer span.End(callReturnTime)
+
+	if err != nil {
+		attrs = append(attrs, labelStatusFailure)
+		span.SetAttributes(labelStatusFailure)
+		span.SetStatus(codes.Error, err.Error())
+	} else {
+		attrs = append(attrs, labelStatusSuccess)
+		span.SetAttributes(labelStatusSuccess)
+		span.SetStatus(codes.Ok, "")
+	}
+
+	if s.spanCorrelation {
+		attrs = appendSpanAndTraceIDFromSpan(attrs, span)
+	}
+
+	s.recorders.operationDuration.Record(
+		spanCtx,
+		float64(time.Since(startTime).Microseconds()),
+		attrs...,
+	)
+	s.counters.operation.Add(ctx, 1, attrs...)
+
+	return output, err
+}
+
+func createCounters(meter metric.Meter) *counters {
+	operationCounter, _ := meter.NewInt64Counter("aws.s3.operation")
```

Comment:
This returned errors should be communicated to the user. Possibly, use the `otel.Handle` function.